src/glyph/parse.rs

```diff
@@ -532,6 +536,73 @@ fn start(reader: &mut Reader<&[u8]>, buf: &mut Vec<u8>) -> Result<GlyphBuilder,
     }
 }
 
+/// Move libs from the lib's `public.objectLibs` into the actual objects.
+/// They key will be removed from the glyph lib.
+fn fill_in_libs(glyph: &mut Glyph) -> Result<(), Error> {
+    if let Some(glyph_lib) = &mut glyph.lib {
+        if let Some(object_libs) = glyph_lib.remove("public.objectLibs") {
+            let object_libs = object_libs
+                .into_dictionary()
+                .ok_or(GlifErrorInternal::Spec { kind: ErrorKind::BadLib, position: 0 })?;
+
+            'next_key: for (key, value) in object_libs.into_iter() {
+                let value = value
+                    .into_dictionary()
+                    .ok_or(GlifErrorInternal::Spec { kind: ErrorKind::BadLib, position: 0 })?;
+
+                if let Some(anchors) = &mut glyph.anchors {
+                    for anchor in anchors {
+                        if let Some(id) = anchor.identifier() {
+                            if id == &key {
+                                anchor.replace_lib(value);
+                                continue 'next_key;
+                            }
+                        }
+                    }
+                }
+                if let Some(guidelines) = &mut glyph.guidelines {
+                    for guideline in guidelines {
+                        if let Some(id) = guideline.identifier() {
+                            if id == &key {
+                                guideline.replace_lib(value);
+                                continue 'next_key;
+                            }
+                        }
+                    }
+                }
+                if let Some(outline) = &mut glyph.outline {
+                    for contour in &mut outline.contours {
+                        if let Some(id) = contour.identifier() {
+                            if id == &key {
+                                contour.replace_lib(value);
+                                continue 'next_key;
+                            }
+                        }
+                        for point in &mut contour.points {
+                            if let Some(id) = point.identifier() {
+                                if id == &key {
+                                    point.replace_lib(value);
+                                    continue 'next_key;
+                                }
+                            }
+                        }
+                    }
+                    for component in &mut outline.components {
+                        if let Some(id) = component.identifier() {
+                            if id == &key {
+                                component.replace_lib(value);
+                                continue 'next_key;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
It's maybe worth noting that this has quadratic runtime; if a glyph contains a large number of items with libs, and a layer contains a large number of such glyphs, we'll end up doing some unnecessary work.

A more robust approach might be to iterate once through each set of items (components, points, etc) and then check if that item's id exists in the object libs map; if it does, take out the lib and assign it to the item. This way we're O(n) over the number of items, instead of O(n*m) over the number of items and the number of object libs.

performance here could be quite bad. If we're worried about worst-case performance, we might want to consider, for each objectlib type, building

Reply:
Hm! Yeah I figured. I'll tinker with this.